src/strategies/silo/SiloStrategy.sol

```diff
@@ -157,15 +188,18 @@ contract SiloStrategy is BaseHealthCheck, TradeFactorySwapper {
         override
         returns (uint256 _totalAssets)
     {
+        silo.accrueInterest(address(asset));
+
         // Only harvest and redeploy if the strategy is not shutdown.
         if (!TokenizedStrategy.isShutdown()) {
-
-            // Check how much we can re-deploy into the yield source.
-            uint256 toDeploy = asset.balanceOf(address(this));
-            // If greater than 0.
-            if (toDeploy > 0) {
-                // Deposit the sold amount back into the yield source.
-                _deployFunds(toDeploy);
+            uint256 _toDeploy = asset.balanceOf(address(this));
+            if (_toDeploy > 0) {
+                uint256 _availableDepositLimit = availableDepositLimit(address(0));
+                if (_toDeploy <= _availableDepositLimit) {
+                    _deployFunds(_toDeploy);
```

Comment:
we also need to be sure `_toDeploy` is not dust amount. If it is a tiny value then this operation will result in "0" and strategy won't be able to report properly.
https://github.com/silo-finance/silo-core-v1/blob/e5d16f201ab2139829d45ed881532c936249d3a5/contracts/lib/EasyMath.sol#L8-L21

**Recommendation:**
Introduce a variable `minAmountToDeposit` and if the amount to be deposited is lesser than this but higher than the available limit then don't deposit anything.

Reply:
Added `DUST_THRESHOLD ` in https://github.com/johnnyonline/yVaults-v3-strategies/pull/8/commits/169b928a4994d0544628eda085a3a7793b9011d4.

Also removed the `Governance2Step` and used `onlyManagement`. Should've done it in a separate commit. Sorry for that.

Reply:
this impl wouldn't fix the issue since `_availableDepositLimit` can still be lesser than `DUST_THRESHOLD` inside the if statement

This should work:
```solidity
if (!TokenizedStrategy.isShutdown()) {
            uint256 _toDeploy = asset.balanceOf(address(this));
            uint256 _availableDepositLimit = availableDepositLimit(address(0));
            if (_toDeploy <= _availableDepositLimit) {
                if (_toDeploy < DUST_THRESHOLD) return; // don't deposit dust
                _deployFunds(_toDeploy);
            } else if (_availableDepositLimit > DUST_THRESHOLD) {
                _deployFunds(_availableDepositLimit);
            }
        }
```

Reply:
Thinking out loud:

hardcoding DUST_THRESHOLD seems ok to me. For USDC that amount would be 0.01 USDC which is fine
for 18 decimal tokens it is small enough so again its ok (rounding error happens when the amount is <10 usually, extremely small)

for WBTC (8 decimals) its 6$ considering 1BTC 60k$. 

So no need to make it storage variable and have a setter for it. Do you see any counterexample token such that this amount is too big?

Reply:
Oh, my bad. Do we want to `return` on `if (_toDeploy < DUST_THRESHOLD)` without calculating `_totalAssets` though?

How does this look https://github.com/johnnyonline/yVaults-v3-strategies/pull/8/commits/c86fdbf9e128d6893e5d15b2f43ebb20df0873bc.

Reply:
Regarding the second comment - yes good point, i saw that [here](https://github.com/fp-crypto/yv3-ajna-summerfi-lev-farming/blob/main/src/Strategy.sol#L71) FP used `100`.

Agree that a setter might be an overkill, as the Strategy's asset will be a stablecoin, most likely? I couldn't think of cases where this would be too big.